cypress/utils/commands.osd.js

```diff
@@ -291,6 +291,41 @@ cy.osd.add('grabDataSourceId', (workspaceName, dataSourceName) => {
   }
 });
 
+cy.osd.add('grabIdsFromDiscoverPageUrl', () => {
+  cy.url().then(($url) => {
+    const workspaceIdMatch = $url.match(/\/w\/([^\/]+)\//);
+    const datasourceIdMatch = $url.match(/dataSource:\(id:'?([0-9a-f-]+)'?,/);
+    const indexPatternIdMatch = $url.match(/\),id:'?([0-9A-Za-z:_-]+)'?,/);
+
+    if (workspaceIdMatch && workspaceIdMatch[1]) {
+      cy.wrap(workspaceIdMatch[1]).as('WORKSPACE_ID');
+    }
+    if (datasourceIdMatch && datasourceIdMatch[1]) {
+      cy.wrap(datasourceIdMatch[1]).as('DATASOURCE_ID');
+    }
+    if (indexPatternIdMatch && indexPatternIdMatch[1]) {
+      cy.wrap(indexPatternIdMatch[1]).as('INDEX_PATTERN_ID');
+    }
+  });
+});
+
+cy.osd.add('deleteAllSavedSearches', (workspaceName) => {
+  cy.osd.navigateToWorkSpaceSpecificPage({
+    workspaceName,
+    page: 'objects',
+    isEnhancement: true,
+  });
+  cy.wait(2000);
+  cy.get('button.euiFilterButton').click();
+  cy.get('.euiFilterSelect__items')
+    .contains(/search/)
+    .click();
+  cy.wait(2000);
+  cy.getElementByTestId('checkboxSelectAll').click();
+  cy.getElementByTestId('savedObjectsManagementDelete').click();
+  cy.getElementByTestId('confirmModalConfirmButton').click();
+});
```

Comment:
i am adding this because before, when using `beforeEach`, we didn't need to worry about deleting the created saved searches since we are starting clean for each test. But with the usage of `before`, this is no longer the case and there are conflicts in the name saving